pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -134,48 +138,49 @@ func pythonDetector(args []string, envs map[string]string) Instrumentation {
 	return None
 }
 
+// isNodeInstrumented parses the provided `os.File` trying to find an
+// entry for APM NodeJS instrumentation. Returns true if finding such
+// an entry, false otherwise.
+func isNodeInstrumented(f *os.File) bool {
+	const readLimit = 16 * 1024 // Read 16KiB max
+
+	limitReader := io.LimitReader(f, readLimit)
+	bufferedReader := bufio.NewReader(limitReader)
+
+	return nodeAPMCheckRegex.MatchReader(bufferedReader)
+}
+
+// nodeDetector checks if a service has APM NodeJS instrumentation.
+//
+// To check for APM instrumentation, we try to find a package.json in
```

Comment:
Maybe some unit tests for this function like Java/Python already have in `detector_nix_test.go` would be useful?